src/api/java/baritone/api/schematic/MirroredSchematic.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.schematic;
+
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.Mirror;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MirroredSchematic implements ISchematic {
+
+    private final ISchematic schematic;
+    private final Mirror mirror;
+
+    public MirroredSchematic(ISchematic schematic, Mirror mirror) {
+        this.schematic = schematic;
+        this.mirror = mirror;
+    }
+
+    @Override
+    public boolean inSchematic(int x, int y, int z, BlockState currentState) {
+        return schematic.inSchematic(
+                mirrorX(x, widthX(), mirror),
+                y,
+                mirrorZ(z, lengthZ(), mirror),
+                mirror(currentState, mirror)
+        );
+    }
+
+    @Override
+    public BlockState desiredState(int x, int y, int z, BlockState current, List<BlockState> approxPlaceable) {
+        return mirror(schematic.desiredState(
```

Comment:
It looks like this flips the schematic in memory each time you call desiredState, I feel as though this could get really laggy on large schematics. Can we instead mirror it once in the constructor?

Reply:
No, because the underlying schematic might have changed since the last call. We can only do this for static schematics, which most of our schematics are not (at least not reliably), and even then we might end up wasting memory by copying a sparse schematic into a dense array.
Also I think baking things in the wrapper schematic itself is actually a bad idea in most cases, because it forces the result to be static while we actually want it to be static iff it wraps a static schematic. Baking should be done separately from constructing the transformed schematic (i.e. `schematic = new BakedSchematic(schematic)`).

IIRC mirroring block states is fairly fast (cache lookup) so the potentially laggy part here is transforming up to 36 states from `approxPlaceable`. Given that we know how `BuilderProcess` uses `desiredState` we could commit a caching crime here (reuse last transformed if last untransformed == new untransformed, assuming the input wasn't mutated).